4-genotype_qc/1-compare_hard_filter_combinations.py

```diff
@@ -254,76 +252,88 @@ def count_tp_fp_t_u(mt_tp: hl.MatrixTable, mt_fp: hl.MatrixTable, mt_syn: hl.Mat
     :param  str var_type: variant type (snv/indel)
     :param str mtdir: matrixtable directory
     :return: Dict containing bin and remaning TP/FP count
-    '''
+    """
     results = {}
 
-    logging.debug(f'Executing count_tp_fp_t_u')
+    logging.debug("Executing count_tp_fp_t_u")
 
     counts = count_tp_fp(mt_tp, mt_fp)
-    results['TP'] = counts[0]
-    results['FP'] = counts[1]
+    results["TP"] = counts[0]
+    results["FP"] = counts[1]
 
-    if var_type == 'snv':
+    if var_type == "snv":
         ratio = get_trans_untrans(mt_syn, pedigree, mtdir)
-        results['t_u_ratio'] = ratio
+        results["t_u_ratio"] = ratio
 
     if mt_prec_recall is not None:
         ht_prec_recall = mt_prec_recall.rows()
 
-        if var_type == 'snv':
-            prec, recall = get_prec_recall(ht_prec_recall, ht_giab, 'snv', mtdir)
-            results['prec'] = prec
-            results['recall'] = recall
+        if var_type == "snv":
+            prec, recall = get_prec_recall(ht_prec_recall, ht_giab, "snv", mtdir)
+            results["prec"] = prec
+            results["recall"] = recall
+        elif var_type == "indel":
+            prec, recall, prec_frameshift, recall_frameshift, prec_inframe, recall_inframe = get_prec_recall(
+                ht_prec_recall, ht_giab, "indel", mtdir
+            )
+            results["prec"] = prec
+            results["recall"] = recall
+            results["prec_inframe"] = prec_inframe
+            results["recall_inframe"] = recall_inframe
+            results["prec_frameshift"] = prec_frameshift
+            results["recall_frameshift"] = recall_frameshift
         else:
-            prec, recall, prec_frameshift, recall_frameshift, prec_inframe, recall_inframe = get_prec_recall(ht_prec_recall, ht_giab, 'indel', mtdir)
-            results['prec'] = prec
-            results['recall'] = recall
-            results['prec_inframe'] = prec_inframe
-            results['recall_inframe'] = recall_inframe
-            results['prec_frameshift'] = prec_frameshift
-            results['recall_frameshift'] = recall_frameshift
-
+            raise ValueError(f"Variant type {var_type} not supported for TP/FP and prec/recall calculations.")
```

Comment:
do we want to raise the error and stop the programme when this happens?

Reply:
Added explanation.